app/controllers/course_summaries_controller.rb

```diff
@@ -46,6 +46,33 @@ def populate
     }
   end
 
+  def grade_distribution
+    table_data = []
+    marking_schemes = current_user.student? ? MarkingScheme.none : MarkingScheme
+
+    marking_schemes.order(id: :asc).each do |m|
+      table_data << m.students_weighted_grade_distribution_array(current_user).data
+    end
+    labels = []
+    (0..19).each do |i|
+      labels << i.to_s + '-' + (i + 1).to_s
+    end
+    datasets = [
+      {
+        label: 'Weighted Total Grades',
```

Comment:
One other comment I just noticed: This should be an internationalized string (using `I18n`). Check how the string is rendered in the existing dashboard page. And it it was hard-coded there too, please add this string to the locale files (under something like `views/course_summary`).